core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -383,6 +456,23 @@ object DocumentRenderer extends Renderer[Document] {
       }
   }
 
+  private lazy val typeExtensionsRenderer: Renderer[List[TypeSystemExtension]] =
+    typeExtensionRenderer.list.contramap(_.sorted)
+
+  private lazy val typeExtensionRenderer: Renderer[TypeSystemExtension] = new Renderer[TypeSystemExtension] {
+    override def unsafeRender(extension: TypeSystemExtension, indent: Option[Int], writer: StringBuilder): Unit =
+      extension match {
+        case t: TypeSystemExtension.SchemaExtension    => schemaExtensionRenderer.unsafeRender(t, indent, writer)
+        case t: TypeExtension.EnumTypeExtension        => enumTypeExtensionRenderer.unsafeRender(t, indent, writer)
+        case t: TypeExtension.InputObjectTypeExtension => inputObjectExtensionRenderer.unsafeRender(t, indent, writer)
+        case t: TypeExtension.InterfaceTypeExtension   => interfaceTypeExtensionRenderer.unsafeRender(t, indent, writer)
+        case t: TypeExtension.ScalarTypeExtension      => () // Scalar can't be extended
```

Comment:
https://spec.graphql.org/October2021/#sec-Scalar-Extensions

> Scalar type extensions are used to represent a scalar type which has been extended from some original scalar type. For example, this might be used by a GraphQL tool or service which adds directives to an existing scalar.

We would need to be able to support adding directives to them for instance